install/outline-install.sh

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2024 community-scripts ORG
+# Author: Gerhard Burger (burgerga)
+# License: MIT
+# https://github.com/community-scripts/ProxmoxVE/raw/main/LICENSE
+
+source /dev/stdin <<< "$FUNCTIONS_FILE_PATH"
+color
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y \
+  sudo \
+  mc \
+  curl \
+  lsb-release \
+  postgresql \
+  gnupg \
+  unzip
+msg_ok "Installed Dependencies"
+
+msg_info "Setting up Node.js Repository"
+mkdir -p /etc/apt/keyrings
+wget -qO- https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
+echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_20.x nodistro main" >/etc/apt/sources.list.d/nodesource.list
+msg_ok "Set up Node.js Repository"
+
+msg_info "Setting up Redis Repository"
+wget -qO- https://packages.redis.io/gpg | gpg --dearmor -o /etc/apt/keyrings/redis-archive-keyring.gpg
+echo "deb [signed-by=/etc/apt/keyrings/redis-archive-keyring.gpg] https://packages.redis.io/deb $(lsb_release -cs) main" >/etc/apt/sources.list.d/redis.list
+msg_ok "Set up Redis Repository"
+
+msg_info "Installing Node.js/Yarn"
+$STD apt-get update
+$STD apt-get install -y nodejs
+$STD npm install -g yarn
+msg_ok "Installed Node.js/Yarn"
+
+msg_info "Installing Redis"
+$STD apt-get install -y redis
+sed -i 's/^bind .*/bind 0.0.0.0/' /etc/redis/redis.conf
+systemctl enable -q --now redis-server.service
+msg_ok "Installed Redis"
+
+msg_info "Setting up PostgreSQL DB"
+SECRET_KEY="$(openssl rand -hex 32)"
+UTILS_SECRET="$(openssl rand -hex 32)"
+DB_NAME=outlinedb
+DB_USER=outline
+DB_PASS="$(openssl rand -base64 18 | tr -d '/' | cut -c1-13)"
+DATABASE_URL="postgres://${DB_USER}:${DB_PASS}@localhost:5432/${DB_NAME}"
+$STD sudo -u postgres psql -c "CREATE ROLE $DB_USER WITH LOGIN PASSWORD '$DB_PASS';"
+$STD sudo -u postgres psql -c "ALTER ROLE $DB_USER WITH CREATEDB;"
+$STD sudo -u postgres psql -c "ALTER ROLE $DB_USER SET timezone TO 'UTC';"
+
+cat <<EOF >~/outline.creds
+Outline-Credentials
+Outline Database User: $DB_USER
+Outline Database Password: $DB_PASS
+Outline Database Name: $DB_NAME
+Outline Secret: $SECRET_KEY
+Outline Utils Secret: $UTILS_SECRET
+EOF
```

Comment:
use echo for this
```bash
{
    echo ""
    echo "Outline-Credentials"
    echo "Outline Database User: $DB_USER"
    echo "Outline Database Password: $DB_PASS"
    echo "Outline Database Name: $DB_NAME"
    echo "Outline Secret: $SECRET_KEY"
    echo "Outline Utils Secret: $UTILS_SECRET"
} >> ~/outline.creds

Reply:
I had echo for this, but then @havardthom told me for another section with echo "use cat <<EOF >/opt/outline/.env instead, see other scripts".
It would really help to have these standards written down somewhere, because for me the logic is not entirely clear...

Reply:
The standards are in the scripts, if you search for `.creds` you see how those sections are normally done, if you search for `.env` you see how those sections are normally done. We would love consistency across sections in a single script, but things have been added over time, so currently consistency across scripts is more important. We normally only comment what we see is inconsistent with other scripts, so no need to change other parts that are not commented.